packages/local-cli/link/linkAll.js

```diff
@@ -0,0 +1,53 @@
+// @flow
+
+import type { ContextT, PlatformsT, ProjectConfigT } from '../core/types.flow';
+
+const log = require('npmlog');
+const { uniqBy, flatten } = require('lodash');
+const path = require('path');
+const getProjectDependencies = require('./getProjectDependencies');
+const getDependencyConfig = require('./getDependencyConfig');
+const promiseWaterfall = require('./promiseWaterfall');
+const commandStub = require('./commandStub');
+const promisify = require('./promisify');
+const linkAssets = require('./linkAssets');
+const linkDependency = require('./linkDependency');
+
+log.heading = 'rnpm-link';
+
+const dedupeAssets = assets => uniqBy(assets, asset => path.basename(asset));
+
+function linkAll(
+  context: ContextT,
+  platforms: PlatformsT,
+  project: ProjectConfigT
+) {
+  log.warn(
+    'Linking modules without specifying package name is deprecated and will be removed in next release'
+  );
+  const dependencies = getProjectDependencies(context.root);
+  const depenendenciesConfig = dependencies.map(dependnecy =>
+    getDependencyConfig(context, platforms, dependnecy)
+  );
+  const assets = dedupeAssets(flatten(depenendenciesConfig.map(d => d.assets)));
```

Comment:
One thing that was there was "project.assets.concat". 

We should also link the project assets. That's the only reason we have a deprecation notice - to make sure those that rely on this behaviour are aware that this is going away.

I wonder whether we should include a notice: 

> "Use react-native link xyz" instead.

and

> If you are using "react-native link" to link assets from your project, let us know here: goo.gl/...

Where goo.gl/... is a link to an issue in this repository. I would like to understand how people are using this feature (and why) and be able to offer them a reasonable workaround.